spec/unit/pops/types/p_object_type_spec.rb

```diff
@@ -0,0 +1,599 @@
+        "expected attribute 'b' to override an inherited attribute")
+    end
+  end
+
+  context 'when dealing with equality' do
+    it 'the attributes can be declared as an array of names' do
+      obj = <<-OBJECT
+        attributes => {
+          a => Integer,
+          b => Integer
+        },
+        equality => [a,b]
+      OBJECT
+      tp = parse_object('MyObject', obj)
+      expect(tp.equality).to eq(['a','b'])
+      expect(tp.equality_attributes.keys).to eq(['a','b'])
+    end
+
+    it 'a single attribute can be declared a name' do
+      obj = <<-OBJECT
+        attributes => {
+          a => Integer,
+          b => Integer
+        },
+        equality => a
+      OBJECT
+      tp = parse_object('MyObject', obj)
+      expect(tp.equality).to eq(['a'])
+    end
+
+    it 'includes all non-constant attributes by default' do
+      obj = <<-OBJECT
+        attributes => {
+          a => Integer,
+          b => { type => Integer, kind => constant, value => 3 },
+          c => Integer
+        }
+      OBJECT
+      tp = parse_object('MyObject', obj)
+      expect(tp.equality).to be_nil
+      expect(tp.equality_attributes.keys).to eq(['a','c'])
+    end
+
+    it 'equalty_include_type is true by default' do
+      obj = <<-OBJECT
+        attributes => {
+          a => Integer
+         },
+        equality => a
+      OBJECT
+      expect(parse_object('MyObject', obj).equality_include_type?).to be_truthy
+    end
+
+    it 'will allow an empty list of attributes' do
+      obj = <<-OBJECT
+        attributes => {
+          a => Integer,
+          b => Integer
+        },
+        equality => []
+      OBJECT
+      tp = parse_object('MyObject', obj)
+      expect(tp.equality).to be_empty
+      expect(tp.equality_attributes).to be_empty
+    end
+
+    it 'will extend default equality in parent' do
+      parent = <<-OBJECT
+        attributes => {
+          a => Integer,
+          b => Integer
+        }
+      OBJECT
+      obj = <<-OBJECT
+        parent => MyObject,
+        attributes => {
+          c => Integer,
+          d => Integer
+        }
+      OBJECT
+      expect_object('MyObject', parent)
+      tp = parse_object('MyDerivedObject', obj)
+      expect(tp.equality).to be_nil
+      expect(tp.equality_attributes.keys).to eq(['a','b','c','d'])
+    end
+
+    it 'will extend equality declared in parent' do
+      parent = <<-OBJECT
+        attributes => {
+          a => Integer,
+          b => Integer
+        },
+        equality => a
+      OBJECT
+      obj = <<-OBJECT
+        parent => MyObject,
+        attributes => {
+          c => Integer,
+          d => Integer
+        }
+      OBJECT
+      expect_object('MyObject', parent)
+      tp = parse_object('MyDerivedObject', obj)
+      expect(tp.equality).to be_nil
+      expect(tp.equality_attributes.keys).to eq(['a','c','d'])
+    end
+
+    it 'will allow that equality contains parent attributes when parent equality does not' do
```

Comment:
"will allow" (when will it? doesn't it do it now?)  use "allows", or "parent defined attributes can be included in equality if not already included by a parent"